### ISSUE TITLE
Error When Compiling on Win10 with Msys2/MinGW

### ISSUE THREAD

**Issue by Jazon22** [ryo-currency/ryo-currency#150]
Getting this error at 32% when compiling.

![image](https://user-images.githubusercontent.com/32424457/49483101-2c766080-f7ef-11e8-8089-5a0acbef3e5b.png)

Any help would be appreciated.

Thanks,
Jason

**Comment by psychocrypt** [comment]
mingw is not supported. please use visual studio or the procompiled
binaries.

**Comment by psychocrypt** [comment]
I removed my last comment because it was wrong.
Do you use mysys64 or 32? Only the 64 nit version is supported.

Am Mi., 5. Dez. 2018, 07:37 hat psychocrypt <psychocrypthpc@gmail.com>
geschrieben:

> mingw is not supported. please use visual studio or the procompiled
> binaries.
>

**Comment by Jazon22** [comment]
msys64 with mingw64.  Thanks.

**Comment by Jazon22** [comment]
After searching for a while I believe this may be the same issue,  

https://github.com/monero-project/monero/issues/4493

Thanks.

**Comment by Jazon22** [comment]
I did update the submodules as well as tried --recursive.  Still got the same error.  I did get it to compile by tweaking configure_checks.cmake but the resulting binary shuts down as soon as I start it.  I'll keep trying! :)

**Comment by psychocrypt** [comment]
could you pease clone it again in a clean folder than compile it and post
all (really all) commands and output to pastbin and link it here.

**Comment by Jazon22** [comment]
Sure thing.  

https://pastebin.com/w4yAD1jV

The tweaks I made to get it to compile are here. https://github.com/Jazon22/ryo-currency/pull/1/commits/9758df2eddcf995cea3a78702840231d86aec9d4

Let me know if you want a log of the compile that fails at 32%.

Thanks for all the help.

**Comment by fireice-uk** [comment]
https://linux.die.net/man/3/evp_md_ctx_cleanup

The problem is with the openssl dependency. Do you have it installed, and if so, what version?

**Comment by Jazon22** [comment]
I do have openssl.  Version 1.1.1a.

![image](https://user-images.githubusercontent.com/32424457/49700441-d8c39880-fb9b-11e8-861d-586683c7caa9.png)

Thanks for the help.

**Comment by Jazon22** [comment]
Searched for another couple hours and I think I need to use an earlier version of openssl.  Something before 1.1, so anything that starts with 1.0.  But I can't figure out how to override.  Any hints on what I should look at?

Thanks.

**Comment by drparadoxical** [comment]
You might not have to, now that I go through the references above, it appears you may simply need to merge in this commit: [monero-project/monero/pull/4471](https://github.com/monero-project/monero/pull/4471).

Once [ryo-currency/ryo-currency/external/unbound/configure_checks.cmake](https://github.com/ryo-currency/ryo-currency/blob/master/external/unbound/configure_checks.cmake) is updated with the merged in changes, try again with openssl v1.1.1.

**Comment by Jazon22** [comment]
@drparadoxical I did try that a few days ago.  I admit this is way out of my wheelhouse but I cloned and then changed  https://github.com/Jazon22/ryo-currency/commit/9758df2eddcf995cea3a78702840231d86aec9d4#diff-8db0e788fee778ca05733296581ec4ab.

I can get it to compile but the executable shuts down as soon as I start it.

Not sure why.  But I do appreciate you giving me more ideas.  

Thanks!

**Comment by drparadoxical** [comment]
That could be something as simple as allowing the daemon to go through the Windows firewall, but who knows though. Windows loves to perform magically and behind several closed doors, bit of a nightmare for those needing to peak behind the curtain in order to do something like develop and debug locally. 

Glad it's compiling now, at least!

**Comment by fireice-uk** [comment]
> I can get it to compile but the executable shuts down as soon as I start it.

Is there any output on the command line? Do all executables behave like that? Does it quit or do you get "this program has stopped working"?

**Comment by Jazon22** [comment]
Hey @fireice-uk.  Unfortunately I uninstalled / deleted everything to start from scratch and can no longer get it to compile.  Now it fails at 73% with the following error.

![image](https://user-images.githubusercontent.com/32424457/50119184-334d9c00-020f-11e9-9f8c-809c628a3045.png)


Thanks for the help.

**Comment by Jazon22** [comment]
I got it to compile again by making this change https://github.com/Jazon22/ryo-currency/commit/6f412e3936a03123581807fb61daafaf6e0bf4e8#diff-aeb85dfdd716bab0b39ef662170d72c4.

But I am back to where I was.  When I try to tun the daemon I get no errors or pop-ups.  It just goes back to the command line.  I tried the wallet and a few other executables and they all act the same.  

![image](https://user-images.githubusercontent.com/32424457/50122726-246ce680-021b-11e9-8b44-5901c808b6e6.png)

Thanks for the help.

**Comment by Jazon22** [comment]
Got a little bit of time to look at this again and found the windows event viewer shows an appcrash.

Here is the log.

Version=1
EventType=APPCRASH
EventTime=131910384668520223
ReportType=2
Consent=1
UploadTime=131910384669617345
ReportStatus=268435456
ReportIdentifier=762cfec6-3135-41e3-85f5-ec923c797617
IntegratorReportIdentifier=ed91e111-56f6-4e91-8ff3-dc37b12d8ce3
Wow64Host=34404
NsAppName=ryod.exe
AppSessionGuid=00006154-0001-0009-1482-26c4cba3d401
TargetAppId=W:0006fcda9e525ae3327bec7c6f4ab7083a1c0000ffff!000013bc880eda54d2bbf239657eee7e76144931ed26!ryod.exe
TargetAppVer=2019//01//03:22:38:51!15d7d0b!ryod.exe
BootId=4294967295
TargetAsId=26697
IsFatal=1
Response.BucketId=a2af2a37b690c36a02fadcc8801c7c22
Response.BucketTable=4
Response.LegacyBucketId=1367648190561287202
Response.type=4
Sig[0].Name=Application Name
Sig[0].Value=ryod.exe
Sig[1].Name=Application Version
Sig[1].Value=0.0.0.0
Sig[2].Name=Application Timestamp
Sig[2].Value=5c2e8efb
Sig[3].Name=Fault Module Name
Sig[3].Value=ryod.exe
Sig[4].Name=Fault Module Version
Sig[4].Value=0.0.0.0
Sig[5].Name=Fault Module Timestamp
Sig[5].Value=5c2e8efb
Sig[6].Name=Exception Code
Sig[6].Value=c0000005
Sig[7].Name=Exception Offset
Sig[7].Value=0000000000643896
DynamicSig[1].Name=OS Version
DynamicSig[1].Value=10.0.17134.2.0.0.256.48
DynamicSig[2].Name=Locale ID
DynamicSig[2].Value=1033
DynamicSig[22].Name=Additional Information 1
DynamicSig[22].Value=026c
DynamicSig[23].Name=Additional Information 2
DynamicSig[23].Value=026cf72cbc55a5bddba946d353d114fc
DynamicSig[24].Name=Additional Information 3
DynamicSig[24].Value=860d
DynamicSig[25].Name=Additional Information 4
DynamicSig[25].Value=860d2a48fac0aa66f4643ffaca4676d1
UI[2]=C:\msys64\home\Jason\ryo-currency\build\release\bin\ryod.exe
LoadedModule[0]=C:\msys64\home\Jason\ryo-currency\build\release\bin\ryod.exe
LoadedModule[1]=C:\WINDOWS\SYSTEM32\ntdll.dll
LoadedModule[2]=C:\WINDOWS\System32\KERNEL32.DLL
LoadedModule[3]=C:\WINDOWS\System32\KERNELBASE.dll
LoadedModule[4]=C:\WINDOWS\System32\ADVAPI32.dll
LoadedModule[5]=C:\WINDOWS\System32\msvcrt.dll
LoadedModule[6]=C:\WINDOWS\System32\sechost.dll
LoadedModule[7]=C:\WINDOWS\System32\RPCRT4.dll
LoadedModule[8]=C:\WINDOWS\System32\SHELL32.dll
LoadedModule[9]=C:\WINDOWS\System32\cfgmgr32.dll
LoadedModule[10]=C:\WINDOWS\SYSTEM32\MSWSOCK.dll
LoadedModule[11]=C:\WINDOWS\SYSTEM32\IPHLPAPI.DLL
LoadedModule[12]=C:\WINDOWS\System32\WS2_32.dll
LoadedModule[13]=C:\WINDOWS\System32\ucrtbase.dll
LoadedModule[14]=C:\WINDOWS\System32\shcore.dll
LoadedModule[15]=C:\WINDOWS\System32\combase.dll
LoadedModule[16]=C:\WINDOWS\System32\bcryptPrimitives.dll
LoadedModule[17]=C:\WINDOWS\System32\windows.storage.dll
LoadedModule[18]=C:\WINDOWS\System32\shlwapi.dll
LoadedModule[19]=C:\WINDOWS\System32\GDI32.dll
LoadedModule[20]=C:\WINDOWS\System32\gdi32full.dll
LoadedModule[21]=C:\WINDOWS\System32\msvcp_win.dll
LoadedModule[22]=C:\WINDOWS\System32\USER32.dll
LoadedModule[23]=C:\WINDOWS\System32\win32u.dll
LoadedModule[24]=C:\WINDOWS\System32\kernel.appcore.dll
LoadedModule[25]=C:\WINDOWS\System32\profapi.dll
LoadedModule[26]=C:\WINDOWS\System32\powrprof.dll
LoadedModule[27]=C:\WINDOWS\System32\FLTLIB.DLL
LoadedModule[28]=C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL
LoadedModule[29]=C:\WINDOWS\System32\IMM32.DLL
LoadedModule[30]=C:\WINDOWS\SYSTEM32\CRYPTSP.dll
LoadedModule[31]=C:\WINDOWS\system32\rsaenh.dll
LoadedModule[32]=C:\WINDOWS\SYSTEM32\bcrypt.dll
State[0].Key=Transport.DoneStage1
State[0].Value=1
OsInfo[0].Key=vermaj
OsInfo[0].Value=10
OsInfo[1].Key=vermin
OsInfo[1].Value=0
OsInfo[2].Key=verbld
OsInfo[2].Value=17134
OsInfo[3].Key=ubr
OsInfo[3].Value=472
OsInfo[4].Key=versp
OsInfo[4].Value=0
OsInfo[5].Key=arch
OsInfo[5].Value=9
OsInfo[6].Key=lcid
OsInfo[6].Value=1033
OsInfo[7].Key=geoid
OsInfo[7].Value=244
OsInfo[8].Key=sku
OsInfo[8].Value=48
OsInfo[9].Key=domain
OsInfo[9].Value=0
OsInfo[10].Key=prodsuite
OsInfo[10].Value=256
OsInfo[11].Key=ntprodtype
OsInfo[11].Value=1
OsInfo[12].Key=platid
OsInfo[12].Value=10
OsInfo[13].Key=sr
OsInfo[13].Value=0
OsInfo[14].Key=tmsi
OsInfo[14].Value=19887
OsInfo[15].Key=osinsty
OsInfo[15].Value=3
OsInfo[16].Key=iever
OsInfo[16].Value=11.472.17134.0-11.0.101
OsInfo[17].Key=portos
OsInfo[17].Value=0
OsInfo[18].Key=ram
OsInfo[18].Value=16226
OsInfo[19].Key=svolsz
OsInfo[19].Value=237
OsInfo[20].Key=wimbt
OsInfo[20].Value=0
OsInfo[21].Key=blddt
OsInfo[21].Value=180410
OsInfo[22].Key=bldtm
OsInfo[22].Value=1804
OsInfo[23].Key=bldbrch
OsInfo[23].Value=rs4_release
OsInfo[24].Key=bldchk
OsInfo[24].Value=0
OsInfo[25].Key=wpvermaj
OsInfo[25].Value=0
OsInfo[26].Key=wpvermin
OsInfo[26].Value=0
OsInfo[27].Key=wpbuildmaj
OsInfo[27].Value=0
OsInfo[28].Key=wpbuildmin
OsInfo[28].Value=0
OsInfo[29].Key=osver
OsInfo[29].Value=10.0.17134.472.amd64fre.rs4_release.180410-1804
OsInfo[30].Key=buildflightid
OsInfo[30].Value=555254D0-6F60-4D6E-9FDF-109C61ADDB8A.1
OsInfo[31].Key=edition
OsInfo[31].Value=Professional
OsInfo[32].Key=ring
OsInfo[32].Value=Retail
OsInfo[33].Key=expid
OsInfo[33].Value=RS:25E7
OsInfo[34].Key=containerid
OsInfo[35].Key=containertype
OsInfo[36].Key=edu
OsInfo[36].Value=0
FriendlyEventName=Stopped working
ConsentKey=APPCRASH
AppName=ryod.exe
AppPath=C:\msys64\home\Jason\ryo-currency\build\release\bin\ryod.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=BE30434F1C44816C0B467906AD038698
MetadataHash=-1091386401

Not sure if it helps any. Makes no sense to me.

Thanks.

**Comment by Jazon22** [comment]
After a ridiculous amount of trial and error I finally got it to compile and run without crashing.  Synch and wallet work great.

Thanks to psychocrypt, fireice-uk, and drparadoxical for trying to help.

Going to close this now.

Thanks.

**Comment by psychocrypt** [comment]
How do you solved it. what was the reason for the first issue.

**Comment by Jazon22** [comment]
There were actually three different issues.  The first caused the compile to fail at 32%, the second caused the compile to fail at 73%.  There are links to those above.  The last issue was after a successful compile the apps would immediately crash (didn't matter which one).

I got them to stop crashing by making a small tweak to CMakeLists.txt, but I still don't know why it worked (doesn't make sense to me).

Let me know if it would be helpful to post a summary of all three fixes.

Thanks.

**Comment by psychocrypt** [comment]
could you please provide the cmake changes. That we can have a look into it
and fix it if it is a bug

**Comment by Jazon22** [comment]
Absolutely.  It would be awesome if I could actually be helpful.  Let me run a few more tests and then will post here.

Thanks.

**Comment by Jazon22** [comment]
@psychocrypt

So I have done this more than a half a dozen times to try to be sure it works.  

To get it compile I had to make these two changes.

https://github.com/Jazon22/ryo-currency/commit/9758df2eddcf995cea3a78702840231d86aec9d4
https://github.com/Jazon22/ryo-currency/commit/6f412e3936a03123581807fb61daafaf6e0bf4e8

To get it to stop crashing on startup I had to re-compile with this change.

https://github.com/Jazon22/ryo-currency/commit/a7824367f4e986f3b7da384a0e896f1b635d0c8f

But I am at a total loss as to why the change to CmakeLists.txt works.  Since I’m running Windows 64 bit I don’t think it should make a difference.  But it definitely does.  So I am stuck with assuming either I don’t understand what the change is doing or there is something weird about my environment.  Or a combination of the two,

So in the end I am not sure if any of this will be of of any use to you.  But if you have any insights as to why it works  it would be much appreciated.

Thanks.
